pontoon/contributors/templates/contributors/includes/timeline.html

```diff
@@ -1,25 +1,46 @@
-<h3>{{ contribution_timeline.title }}</h3>
-
-{% for title, values in contribution_timeline.contributions.items() %}
+{% macro render_contribution_group(title, contribution_type, entries) %}
 <div class="contribution-group">
     <h4 class="heading">
-        <span class="icon far {{ values.type }}"></span>
+        <span class="icon far {{ contribution_type }}"></span>
         {{ title }}
     </h4>
     <ul class="localizations">
-        {% for _, localization in values.data.items() %}
+        {% for localization in entries %}
         <li>
             <a href="{{ localization.url }}">
-                <span class="stress">{{ localization.project.name }}</span> &middot; {{
-                localization.locale.name }} <span class="stress">{{ localization.locale.code }}</span>
-                <span class="contribution-count">{{ ", ".join(localization.actions) }}</span>
+                <span class="stress">{{ localization.project.name }}</span> &middot;
+                {{ localization.locale.name }} <span class="stress">{{ localization.locale.code }}</span>
+                
+                {% if contribution_timeline.show_year %}
+                    <span class="contribution-count">{{ ", ".join(localization.actions) }}</span>
+                {% else %}
+                    <span class="contribution-count">{{ ", ".join(localization.action_dates) }}</span>
+                {% endif %}
+
             </a>
         </li>
         {% endfor %}
     </ul>
 </div>
-{% endfor %}
+{% endmacro %}
+
+<h3>{{ contribution_timeline.title }}</h3>
 
 {% if contribution_timeline.contributions == {} %}
-<p>No activity in this period.</p>
+    <p>No activity in this period.</p>
 {% endif %}
+
+{% if contribution_timeline.show_year %}
+    {% for month, contribution_types in contribution_timeline.contributions.items() %}
+        <div class="month-group">
+            <h4>{{ month }}</h4>
+            {% for contribution_type, details in contribution_types.items() %}
+                {{ render_contribution_group(details.title, contribution_type, details.entries) }}
+            {% endfor %}
+        </div>
+    {% endfor %}
+{% else %}
+    {% for title, project_data in contribution_timeline.contributions.items() %}
```

Comment:
I find it confusing that we have different "APIs" for this month's data and for the full year data. I would expect the backend functions to return data in the exact same format, but for different periods, so we wouldn't need to IF-ELSE here and in one or two other places.